scripts/zones/Oldton_Movalpolos/npcs/Twinkbrix.lua

```diff
@@ -6,48 +6,117 @@
 -----------------------------------
 local entity = {}
 
+entity.onTrigger = function(player, npc)
+    -- Trigger is only ever informative events about key item and cooldown status
+    local operatingLeverCD = player:getCharVar('[ENM]OperatingLever')
+    local operatingLeverWaiting = operatingLeverCD > VanadielTime()
+    local operatingLeverHas = player:hasKeyItem(xi.ki.SHAFT_2716_OPERATING_LEVER)
+
+    local gateDialCD = player:getCharVar('[ENM]GateDial')
+    local gateDialWaiting = gateDialCD > VanadielTime()
+    local gateDialHas = player:hasKeyItem(xi.ki.SHAFT_GATE_OPERATING_DIAL)
+
+    if
+        operatingLeverHas and
+        gateDialHas
+    then
+        player:startEvent(50)
+
+    -- lever on cooldown
+    else
+        -- generic messages about how to obtain both items
+        -- player:startEvent(52, 1781)
+
+        local eventID = (gateDialHas and operatingLeverWaiting) and 53 or 52
+        local operatingLeverParam = operatingLeverWaiting and operatingLeverCD or 432000
+        local gateDialParam = gateDialWaiting and gateDialCD or 432000
+        local keyItemFlag = 0 -- This event param passes whether the player has/can get both key items
+        keyItemFlag = utils.mask.setBit(keyItemFlag, 0, operatingLeverWaiting)
+        keyItemFlag = utils.mask.setBit(keyItemFlag, 1, gateDialWaiting)
+        keyItemFlag = utils.mask.setBit(keyItemFlag, 2, operatingLeverHas)
+        keyItemFlag = utils.mask.setBit(keyItemFlag, 3, gateDialHas)
+
+        player:startEvent(eventID, 1781, operatingLeverParam, 2964, gateDialParam, keyItemFlag)
+    end
+end
+
 entity.onTrade = function(player, npc, trade)
+    local operatingLeverCD = player:getCharVar('[ENM]OperatingLever')
+    local gateDialCD = player:getCharVar('[ENM]GateDial')
+    local specialGilTrade = 2716
     local mineShaftWarpCost = 2000
     local tradeGil = trade:getGil()
 
     if
         player:hasKeyItem(xi.ki.SHAFT_GATE_OPERATING_DIAL) and
-        npcUtil.tradeHas(trade, { { 'gil', mineShaftWarpCost } })
+        npcUtil.tradeHasExactly(trade, { { 'gil', mineShaftWarpCost } })
     then
-        player:startEvent(56)
+        -- teleport for mineShaftWarpCost relies on having SHAFT_GATE_OPERATING_DIAL
+        -- but consumes SHAFT_2716_OPERATING_LEVER (after confirming with you)
+        if player:hasKeyItem(xi.ki.SHAFT_2716_OPERATING_LEVER) then
+            player:startEvent(56, 1781, 23, 1757, 177552692, 8, 17407, 15, 0)
```

Comment:
This looks like an NPC ID, can you confirm it's either you and or/ the goblin? `177552692`

Reply:
npc id is 16822535

![image](https://github.com/LandSandBoat/server/assets/131182600/8a7abc91-d3ee-4003-afd8-9a8987da862c)

Looks like it works fine without the rest of the params. Want to nix them?

Reply:
The first parameter after event ID is an item id. Concretely, the item you defined.

Reply:
Sure seems that way.

Reply:
Updated